app/controllers/changeFinancialInstitution/ChangeFinancialInstitutionController.scala

```diff
@@ -0,0 +1,108 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.changeFinancialInstitution
+
+import com.google.inject.Inject
+import controllers.actions._
+import models.requests.DataRequest
+import models.{ChangeAnswers, UserAnswers}
+import pages.changeFinancialInstitution.ChangeFiDetailsInProgress
+import play.api.Logging
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.{FinancialInstitutionUpdateService, FinancialInstitutionsService}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import utils.ContactHelper
+import viewmodels.changeFinancialInstitution.ChangeFinancialInstitutionViewModel.getChangeFinancialInstitutionSummaries
+import viewmodels.common.{getFirstContactSummaries, getSecondContactSummaries}
+import viewmodels.govuk.summarylist._
+import views.html.ThereIsAProblemView
+import views.html.changeFinancialInstitution.ChangeFinancialInstitutionView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ChangeFinancialInstitutionController @Inject() (
+  override val messagesApi: MessagesApi,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  financialInstitutionsService: FinancialInstitutionsService,
+  financialInstitutionUpdateService: FinancialInstitutionUpdateService,
+  val controllerComponents: MessagesControllerComponents,
+  view: ChangeFinancialInstitutionView,
+  errorView: ThereIsAProblemView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with ContactHelper
+    with I18nSupport
+    with Logging {
+
+  def onPageLoad(fiId: String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      val userAnswers = request.userAnswers
+      financialInstitutionsService
+        .getFinancialInstitution(request.fatcaId, fiId)
+        .flatMap {
+          case Some(fiDetails) =>
+            userAnswers.get(ChangeFiDetailsInProgress) match {
+              case Some(true) =>
+                val hasChanges = financialInstitutionUpdateService.fiDetailsHasChanged(userAnswers, fiDetails)
+                Future.successful(createPage(fiId, userAnswers, hasChanges))
+              case _ =>
+                financialInstitutionUpdateService
+                  .populateAndSaveFiDetails(userAnswers, fiDetails)
+                  .map(createPage(fiId, _, hasChanges = false))
+                  .recoverWith {
+                    exception =>
+                      logger.error(s"Failed to populate and save FI details to user answers for: [$fiId]", exception)
+                      Future.successful(InternalServerError(errorView()))
+                  }
```

Comment:
For our services we use a custom error page instead of the default. I believe there's been a couple of tickets to fix instances where the default error was returned.